File: 19. Remove Nth Node From End of List/solution.js
function removeNthFromEnd(head, n) {
    let start = head;

    let count = 0;
    while (count < n && head !== null) {
        head = head.next
        count++;
    }

    let prev = undefined;
    let second = start;

    while (head !== null) {
        prev = second;
        second = second.next;
        head = head.next;
    }

    if (prev !== undefined) {
        prev.next = second.next;
        second.next = undefined;
    } else {
        start = second.next;
        second.next = undefined;
    }

    return start;
}

File: 19. Remove Nth Node From End of List/solution.go
package main

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	start := head

	count := 0

	for count < n && head != nil {
		head = head.Next
		count++
	}

	var prev *ListNode
	second := start
	for head != nil {
		prev = second
		head = head.Next
		second = second.Next
	}

	if prev != nil {
		prev.Next = second.Next
		second.Next = nil
	} else {
		start = second.Next
		second.Next = nil
	}

	return start
}
